src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala

```diff
@@ -95,22 +111,55 @@ case class GetManyDocumentsQueryBuilder(client: Client,
     queryByIds[CouchKeyVals[String, CouchDocRev]](ids, params)
   }
 
-  def queryAllowMissing(ids: Seq[String]): Task[CouchKeyValsIncludesMissing[String, CouchDocRev]] = {
+  def queryAllowMissing(ids: Seq[String]):
+  Task[CouchKeyValsIncludesMissing[String, CouchDocRev]] = {
     queryByIds[CouchKeyValsIncludesMissing[String, CouchDocRev]](ids, params)
   }
 
+  @deprecated(
+               "Fails if different document types exist in the Db. " +
+               "Use `queryByTypeIncludeDocs[D]` instead", "0.7.1")
   def queryIncludeDocs[D: R]: Task[CouchDocs[String, CouchDocRev, D]] = {
     queryWithoutIds[CouchDocs[String, CouchDocRev, D]](includeDocs().params)
   }
 
+  def queryByTypeIncludeDocs[K, V, D: R](typeFilterView: CouchView)
+                                        (implicit tag: ClassTag[D], kr: R[K], kw: W[K], vr: R[V]):
+  Task[CouchDocs[K, V, D]] = {
+    typeMappings.forType(tag.runtimeClass) match {
+      case Some(k) => queryByType[K, V, D](typeFilterView, k)
+      case None => Res.Error("not_found", s"type mapping not found").toTask[CouchDocs[K, V, D]]
+    }
+  }
+
+  def queryByTypeIncludeDocs[D: R](implicit tag: ClassTag[D]):
```

Comment:
What do you think about calling this something like queryByTypeIncludeDocsWithTemporaryView? :) It's kind of long, but it makes the difference with the other queryByTypeIncludeDocs clearer.